test/integration/controller/job/job_controller_test.go

```diff
@@ -1027,4 +1028,76 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", func() {
 		})
 	})
 
+	ginkgo.It("Should schedule jobs when partial admission is enabled", func() {
+		origPartialAdmission := features.Enabled(features.PartialAdmission)
```

Comment:
no,  you need a testing.TB which cannot be implemented (it has a private field) a workaround could be to pass nil but if something goes wrong we'll end up with a more or less cryptic panic instead of an error message.

The right way, in my opinion wold be to change `component-base`  so it will a  narrower interface, something like:
```diff
- func SetFeatureGateDuringTest(tb testing.TB, gate featuregate.FeatureGate, f featuregate.Feature, value bool) func() {
+ type errorer interface {
+    	Errorf(format string, args ...any)
+}
+ func SetFeatureGateDuringTest(tb errorer, gate featuregate.FeatureGate, f featuregate.Feature, value bool) func() {
```

After this we can use ```go 	ginkgo.GinkgoT() ```

Should I try my luck with a PR there?

Reply:
It looks like GinkgoT returns an object compatible with TB, so you should be able to pass it.

Reply:
I'll open do followup for this if/when  kubernetes/kubernetes#118346 is merged.